book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+- When Integration service sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating new “ephemeral” StoneSoup Environment CRs which trigger the GitOps Service to provision environment.
+- After the environment is ready and the application snapshot has been deployed to it, the integration service tests and validates the application according to user-provided configuration. It does this by executing Tekton Pipelineruns against the ephemeral Environment.
+- When all of the test Tekton PipelineRuns have passed, Integration service triggers a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
+- Finally, if any automatic ReleasePlans have been created by the user in the workspace, it will create a Release CR signalling intent to release the tested content, to be carried out by the [release-service](./release-service.md).
+
+The diagram below shows the interaction of the integration service and other services.
+
+![](../diagrams/hacbs-data-flow.jpg)
+
+### Dependencies
+
+The [Integration Service](./integration-service.md) is dependent on the following services:
+- [Pipeline Service](./pipeline-service.md)
+  - Pipeline execution, Pipeline logging
+- [GitOps Service](./gitops-service.md)
+  - Provides the facility to create
+    - Snapshots defining sets of Builds to test
+    - Environment to test the Application on
+    - SnapshotEnvironmentBindings to have the Snapshot of the Application deployed to a specific Environment
+- [Hybrid Application Service](./hybrid-application-service.md)
+  - Provides the Application and Component model. Integration Service updates the pullspec reference on the Component CR when a component passes its required testing.
+- [Release Service](./release-service.md)
+  - Provides the ReleasePlan that will be used to determine if integration-service should create a Release
+- [Enterprise Contract Service](./enterprise-contract.md)
+  - Provides facilities to validate whether content has passed the Enterprise Contract.
+
+
+## Terminology
+
+**Component Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline.
+
+**Composite Pipeline** - This is the pipeline run that runs if two or more Components have completed and passed their Component pipeline concurrently so that they should be tested together.
+
+**Global Candidate List** - The list of all Component digest that have passed individual component testing. This can be retrieved from the Application to see what Components it is made of and then querying each of the Components `spec.containerImage`. 
+
+**Snapshot** - The list that maps all Components of an Application to their Component Image digests. Once created, the snapshot is immutable. 
```

Comment:
Would `Once created, the list of Components with their images is immutable.` work?

Reply:
I just get hung up where we state that it is generally immutable here and then later mutate part of it that we never mention until that point.

Reply:
Hm, the fact that the `Snapshot` is an k8s custom resource may be missing here to explain the fact that it can have a status field even if the contents of its spec are immutable. 

Something like this could clear that up:
```
**Snapshot** - The custom resource that contains the list of all Components of an Application with their Component Image digests. Once created, the list of Components with their images is immutable. The Integration service updates the status of the resource to reflect the testing outcome.
```

Reply:
That sounds clear to me even if you are not familiar with custom resources having a status!

Reply:
Done